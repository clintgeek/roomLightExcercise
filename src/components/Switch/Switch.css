.powerToggle {}

.powerToggleLabel:first-of-type {
  margin-right: 1.5rem;
}