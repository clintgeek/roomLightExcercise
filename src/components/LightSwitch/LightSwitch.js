import React from 'react';
import Switch from '@material-ui/core/Switch';

function LightSwitch(props) {
  const togglePower = () => {
    props.setPower(!props.powered);
  };

  return (
    <div className="Switch">
      <Switch
        checked={props.powered}
        onChange={togglePower}
        name="power"
        inputProps={{ 'aria-label': 'power' }}
      />
    </div>
  );
}

export default LightSwitch;