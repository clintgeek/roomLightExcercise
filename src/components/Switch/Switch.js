import React from 'react';
import './Switch.css';

function Switch(props) {
  const togglePower = () => {
    props.setPower(!props.powered);
  };


  return (
    <div className="Switch">
      <label className="powerToggleLabel">
        <input
          type="radio"
          name="power-on"
          value={true}
          checked={props.powered}
          onChange={togglePower}
          className="powerToggle"
        />
        ON
      </label>
      <label className="powerToggleLabel">
        <input
          type="radio"
          name="power-off"
          value={true}
          checked={!props.powered}
          onChange={togglePower}
          className="powerToggle"
        />
        OFF
      </label>
    </div>
  );
}

export default Switch;