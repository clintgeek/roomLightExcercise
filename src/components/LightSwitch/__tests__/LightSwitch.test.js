import React from 'react';
import {shallow} from 'enzyme';
import LightSwitch from '../LightSwitch';
import Switch from '@material-ui/core/Switch';

describe('<LightSwitch> component', () => {
  let wrapper;
  const togglePowerMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<LightSwitch  powered={false} togglePower={togglePowerMock} />);
  })

  it('should match snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();  
  });

  it('should render a <Switch> with expected props', () => {
    const expectedProps = {
      'checked': false,
      'inputProps': {
        'aria-label': 'power',
      },
      'name': 'power',
      'onChange': togglePowerMock,
    }
    const switchComponent = wrapper.find(Switch);
    expect(switchComponent.isEmptyRender()).toBe(false);
    expect(switchComponent.props()).toEqual(expectedProps)
  });

  it('should call props.togglePower when changed', () => {
    wrapper.find(Switch).props().onChange();
    expect(togglePowerMock).toHaveBeenCalledTimes(1);
  });

  it('should match its checked property to props.powered', () => {
    const isCheckedWithFalse = wrapper.find(Switch).props().checked;
    expect(isCheckedWithFalse).toBe(false);
    
    const checkedWrapper = shallow(<LightSwitch powered={true} togglePower={togglePowerMock} />)
    const isCheckedWithTrue = checkedWrapper.find(Switch).props().checked;
    expect(isCheckedWithTrue).toBe(true);
  })
});