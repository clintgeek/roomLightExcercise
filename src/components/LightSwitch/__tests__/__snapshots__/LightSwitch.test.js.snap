// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LightSwitch> component should match snapshot 1`] = `
"<div>
  <WithStyles(ForwardRef(Switch)) checked={false} onChange={[Function: mockConstructor]} name=\\"power\\" inputProps={{...}} />
</div>"
`;
